sapi/fpm/fpm/fpm_php.c

```diff
@@ -114,19 +116,32 @@ int fpm_php_apply_defines_ex(struct key_value_s *kv, int mode) /* {{{ */
 static int fpm_php_apply_defines(struct fpm_worker_pool_s *wp) /* {{{ */
 {
 	struct key_value_s *kv;
+	int apply_result;
+	bool extension_loaded = false;
 
 	for (kv = wp->config->php_values; kv; kv = kv->next) {
-		if (fpm_php_apply_defines_ex(kv, ZEND_INI_USER) == -1) {
+		apply_result = fpm_php_apply_defines_ex(kv, ZEND_INI_USER);
+		if (apply_result == -1) {
 			zlog(ZLOG_ERROR, "Unable to set php_value '%s'", kv->key);
+		} else if (apply_result == 2) {
+			extension_loaded = true;
 		}
 	}
 
 	for (kv = wp->config->php_admin_values; kv; kv = kv->next) {
-		if (fpm_php_apply_defines_ex(kv, ZEND_INI_SYSTEM) == -1) {
+		apply_result = fpm_php_apply_defines_ex(kv, ZEND_INI_SYSTEM);
+		if (apply_result == -1) {
 			zlog(ZLOG_ERROR, "Unable to set php_admin_value '%s'", kv->key);
+		} else if (apply_result == 2) {
+			extension_loaded = true;
 		}
 	}
 
+	if (extension_loaded) {
+		zend_destroy_module_handlers();
+		zend_collect_module_handlers();
```

Comment:
To avoid the API break we may call only ``zend_collect_module_handlers()`` and make it take care about freeing old ``class_cleanup_handlers`` and ``module_request_startup_handlers``.

Reply:
So I just use `realloc` instead of `malloc` for `class_cleanup_handlers` and `module_request_startup_handlers` which I think should do the job.

Reply:
hmm so this broke phpdbg for some reason so will need to check that out...

Reply:
ah it runs collect after destroy - doing multiple startups. So the value needs to be set to NULL after freeing to make this work.